### DISCUSSION TITLE
How can I make a `ui.table` responsive using quasar's `dense` or `grid` prop?

### DISCUSSION THREAD

**Discussion by tmlmt** [zauberzeug/nicegui#3160] [Q&A] [answered]
### Question

How can I turn a wide table to show as grid on mobile devices, using the [`grid` prop](https://quasar.dev/vue-components/table#responsive-tables) of quasar? 

`ui.table(...).props(":grid")` works but `ui.table(...).props(":grid='$q.screen.lt.md'")` does not.

**Comment by falkoschindler** [comment]
Hi @tmlmt,
I'm not sure if responsive props are possible at all. A rather hacky workaround could be to generate two tables, one with "grid" and one without, and to use responsive Tailwind classes to toggle visibility like `.classes('lg:hidden')`.

**Reply by tmlmt** [reply]
Hmm ok :// I was thinking:
- could we introduce a vue component for the quasar element, so that the screen plugin can be used as well? Like it's done for the plotly component?
- or is there any way in nicegui (via fastapi?) to detect window size and add an event listener?

**Reply by falkoschindler** [reply]
A resize event listener could look like this:
```py
ui.add_head_html('<script>window.addEventListener("resize", () => emitEvent("resize"))</script>')
ui.on('resize', lambda: ui.notify('Window resized'))
```

**Comment by CrystalWindSnake** [comment, marked as answer]
It works:

```python
ui.table(columns=columns, rows=rows).props(':grid="Quasar.Screen.lt.md"')
```

**Reply by tmlmt** [reply]
Amazing ! Thanks a lot ! So the trick was `Quasar.Screen` instead of `$q.screen` :+1: 
I'll add that to my documentation PR that I'll work on soon, cf https://github.com/zauberzeug/nicegui/discussions/3155

**Reply by falkoschindler** [reply]
Wow! I didn't know `Quasar.Screen` and I wouldn't have expected it to be responsive, i.e. being re-evaluated whenever the screen size changes. What a powerful solution! 🚀

**Reply by CrystalWindSnake** [reply]
![image](https://github.com/zauberzeug/nicegui/assets/36034954/fd134da0-4601-401e-8a4d-7f6750eef477)
Thanks to NiceGUI's simple architecture, we almost get all the functionalities of Quasar. By pressing F12 in the browser and entering Quasar in the console, you can access all available functional objects.
